### PR TITLE
[OCPBUGS-35431] Add SNO installation method for LPAR IBM Z

### PR DESCRIPTION
**Pull request by SNiemann15**
Version(s): 4.16+

Issue: https://issues.redhat.com/browse/OCPBUGS-35431

Link to docs preview:
https://77425--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_sno/install-sno-installing-sno.html#install-sno-with-ibmz

QE review:
- [x] QE has approved this change.

Additional information:
https://docs.google.com/document/d/1zrXtjF3FoRduLh13jts568O-TxuMjMJqbz9OSWi86tA/edit#heading=h.q48izkry3x13

### CODE REVIEW

File: modules/install-sno-ibm-z-lpar.adoc [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+//
+// installing_sno/install-sno-installing-sno.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="installing-sno-on-ibm-z-lpar_{context}"]
+= Installing {sno} in an LPAR on {ibm-z-title} and {ibm-linuxone-title}
+
+.Prerequisites
+
+* You have installed `podman`.
```
Comment (smolin-de) [inline modules/install-sno-ibm-z-lpar.adoc:11]: Why do we need podman ?
I would remove this line, until it is required.

File: modules/install-sno-ibm-z-lpar.adoc [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+* You have an existing `install-config.yaml` file.
```
Comment (smolin-de) [inline modules/install-sno-ibm-z-lpar.adoc:12]: Why do we require to have an existing "install-config.yaml" ?
Because, It is described below how to "Prepare the `install-config.yaml` file"
I would remove this line.

File: modules/install-sno-ibm-z-lpar.adoc [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+----
+<1> For the `coreos.live.rootfs_url=` artifact, specify the matching `rootfs` artifact for the `kernel`and `initramfs` you are booting. Only HTTP and HTTPS protocols are supported.
+<2> For the `ignition.config.url=` parameter, specify the Ignition file for the machine role. Only HTTP and HTTPS protocols are supported.
+<3> For the `ip=` parameter, assign the IP address automatically using DHCP or manually as described in "Installing a cluster with z/VM on {ibm-z-name} and {ibm-linuxone-name}".
+<4> For installations on DASD-type disks, use `rd.dasd=` to specify the DASD where {op-system} is to be installed. Omit this entry for FCP-type disks.
+<5> For installations on FCP-type disks, use `rd.zfcp=<adapter>,<wwpn>,<lun>` to specify the FCP disk where {op-system} is to be installed. Omit this entry for DASD-type disks.
++
+Leave all other parameters unchanged.
+
+. Transfer the following artifacts, files, and images to z/VM. For example by using FTP:
```
Comment (sandisamp) [inline modules/install-sno-ibm-z-lpar.adoc:180]: It should be LPAR, not z/VM

File: modules/install-sno-ibm-z-lpar.adoc [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+** `kernel` and `initramfs` artifacts
+** Parameter files
+** {op-system} images
++
+For details about how to transfer the files with FTP and boot, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/performing_a_standard_rhel_8_installation/installing-in-an-lpar_installing-rhel[Installing in an LPAR].
+
+. Boot the machine
```
Comment (sandisamp) [inline modules/install-sno-ibm-z-lpar.adoc:188]: Since this is a process with bootstrap separate, we have to boot both the bootstrap and then subsequently the SNO LPAR (or the master) 

Consider this line here instead. 

```
Boot the bootstrap and master machine with their respective ignition config and boot params.
```

File: modules/install-sno-ibm-z-lpar.adoc [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+.Prerequisites
+
+* If you are deploying a single-node cluster there are zero compute nodes; the Ingress Controller pods run on the control plane nodes. In single-node cluster deployments, you must configure your application ingress load balancer to route HTTP and HTTPS traffic to the control plane nodes. See the _Load balancing requirements for user-provisioned infrastructure_ section for more information.
+* When you create the Kubernetes manifest files, ensure that the `mastersSchedulable` parameter in the `<installation_directory>/manifests/cluster-scheduler-02-config.yml` file is set to `true`, as shown in the following `spec` stanza:
++
+[source,yaml]
+----
+spec:
+  mastersSchedulable: false
```
Comment (sandisamp) [inline modules/install-sno-ibm-z-lpar.adoc:17]: Shouldn't this value be shown as true? We are telling the users to set the value to true

Comment (SNiemann15) [reply modules/install-sno-ibm-z-lpar.adoc]: True 👍

File: modules/install-sno-ibm-z-lpar.adoc [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+ip=<ip>::<gateway>:<netmask>:<hostname>::none nameserver=<dns> \// <3>
+rd.znet=qeth,0.0.1140,0.0.1141,0.0.1142,layer2=1,portno=0 \
+rd.dasd=0.0.4411 \// <4>
+rd.zfcp=0.0.8001,0x50050763040051e3,0x4000406300000000 \// <5>
+zfcp.allow_lun_scan=0
+----
+<1> Specify the location of the `master.ign` config file. Only HTTP and HTTPS protocols are supported.
+<2> For the `coreos.live.rootfs_url=` artifact, specify the matching `rootfs` artifact for the `kernel`and `initramfs` you are booting. Only HTTP and HTTPS protocols are supported.
+<3> For the `ip=` parameter, assign the IP address manually as described in "Installing a cluster in an LPAR on {ibm-z-name} and {ibm-linuxone-name}".
+<4> For installations on DASD-type disks, use `rd.dasd=` to specify the DASD where {op-system} is to be installed. Omit this entry for FCP-type disks.
```
Comment (smolin-de) [inline modules/install-sno-ibm-z-lpar.adoc:192]: In this example we boot from FC disk "install_dev"=/dev/sda.
```suggestion
<4> For installations on DASD-type disks, use `rd.dasd=` to specify the DASD where {op-system} is to be installed. Change the 'install_dev' parameter to 'coreos.inst.install_dev=/dev/dasda' and omit this entry for FCP-type disks.
```

File: modules/install-sno-ibm-z-lpar.adoc [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+
+. Transfer the following artifacts, files, and images to the LPAR. For example by using FTP:
+
+** `kernel` and `initramfs` artifacts
+** Parameter files
+** {op-system} images
++
+For details about how to transfer the files with FTP and boot, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/performing_a_standard_rhel_8_installation/installing-in-an-lpar_installing-rhel[Installing in an LPAR].
+
+. Boot the machine.
```
Comment (smolin-de) [inline modules/install-sno-ibm-z-lpar.adoc:225]:
```suggestion
. Boot the bootstrap machine.
. Boot the control plane machine.
```

File: modules/install-sno-ibm-z-lpar.adoc (hunk 4 above)
Comment (SNiemann15) [reply modules/install-sno-ibm-z-lpar.adoc]: Discussed with @smolin-de that "Boot the machine" is sufficient.

**Review by sandisamp — APPROVED** [review]

**Review by smolin-de — APPROVED** [review]

**Comment by holgwolf** [comment]
LGTM

File: modules/install-sno-ibm-z-lpar.adoc [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+.Procedure
+
+. Set the {product-title} version by running the following command:
++
+[source,terminal]
+----
+$ OCP_VERSION=<ocp_version> <1>
+----
++
+<1> Replace `<ocp_version>` with the current version, for example, `latest-{product-version}`
```
Comment (obrown1205) [inline modules/install-sno-ibm-z-lpar.adoc:22]: Needs punctuation:

<1> Replace `<ocp_version>` with the current version. For example, use `latest-{product-version}`.

File: modules/install-sno-ibm-z-lpar.adoc [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+
+. Change to the directory that contains the {product-title} installation program and generate the Kubernetes manifests for the cluster:
++
+[source,terminal]
+----
+$ ./openshift-install create manifests --dir <installation_directory> <1>
+----
++
+<1> For `<installation_directory>`, specify the installation directory that
+contains the `install-config.yaml` file you created.
```
Comment (obrown1205) [inline modules/install-sno-ibm-z-lpar.adoc:124]: You can remove the line break and include the full sentence on line 123, unless there was a specific reason to add a line :)

**Comment by obrown1205** [comment]
Just a couple things, otherwise LGTM! :) Great work!

File: modules/install-sno-ibm-z-lpar.adoc (hunk 8 above)
Comment (SNiemann15) [reply modules/install-sno-ibm-z-lpar.adoc]: These modules have the same sentence I'll add them to the PR
modules/install-sno-ibm-z-kvm.adoc
modules/install-sno-ibm-z.adoc

File: installing/installing_sno/install-sno-installing-sno.adoc [hunk 10]
```diff
@@ -139,7 +139,11 @@ include::modules/creating-custom-live-rhcos-iso.adoc[leveloffset=+1]
 [id="install-sno-with-ibmz"]
 == Installing {sno} with {ibm-z-title} and {ibm-linuxone-title}
 
-Installing a single-node cluster on {ibm-z-name} and {ibm-linuxone-name} requires user-provisioned installation using either the "Installing a cluster with {op-system-base} KVM on {ibm-z-name} and {ibm-linuxone-name}" or the "Installing a cluster with z/VM on {ibm-z-name} and {ibm-linuxone-name}" procedure.
+Installing a single-node cluster on {ibm-z-name} and {ibm-linuxone-name} requires user-provisioned installation using one of the following procedures:
+
+* Installing a cluster with z/VM on {ibm-z-name} and {ibm-linuxone-name}
+* Installing a cluster with {op-system-base} KVM on {ibm-z-name} and {ibm-linuxone-name}
+* Installing a cluster in an LPAR on {ibm-z-name} and {ibm-linuxone-name}
```
Comment (jeana-redhat) [inline installing/installing_sno/install-sno-installing-sno.adoc:146]: Since this is an assembly, wdyt about making these into xrefs so the user can just click what they want?

File: installing/installing_sno/install-sno-installing-sno.adoc [hunk 11]
```diff
@@ -161,12 +165,15 @@ You can use dedicated or shared IFLs to assign sufficient compute resources. Res
 .Additional resources
 
 * xref:../../installing/installing_ibm_z/installing-ibm-z.adoc#installing-ibm-z[Installing a cluster with z/VM on {ibm-z-name} and {ibm-linuxone-name}]
-* xref:../../installing/installing_ibm_z/installing-ibm-z-kvm.adoc#installing-ibm-z-kvm[Installing a cluster with {op-system-base} KVM on {ibm-z-name} and{ibm-linuxone-name}]
+* xref:../../installing/installing_ibm_z/installing-ibm-z-kvm.adoc#installing-ibm-z-kvm[Installing a cluster with {op-system-base} KVM on {ibm-z-name} and {ibm-linuxone-name}]
+* xref:../../installing/installing_ibm_z/installing-ibm-z-lpar.adoc#installing-ibm-z-lpar[Installing a cluster in an LPAR on {ibm-z-name} and {ibm-linuxone-name}]
```
Comment (jeana-redhat) [inline installing/installing_sno/install-sno-installing-sno.adoc:169]: If you choose to link as suggested above, could remove this section

File: modules/install-sno-ibm-z-lpar.adoc [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+//
+// installing_sno/install-sno-installing-sno.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="installing-sno-on-ibm-z-lpar_{context}"]
+= Installing {sno} in an LPAR on {ibm-z-title} and {ibm-linuxone-title}
+
+.Prerequisites
+
+* If you are deploying a single-node cluster there are zero compute nodes; the Ingress Controller pods run on the control plane nodes. In single-node cluster deployments, you must configure your application ingress load balancer to route HTTP and HTTPS traffic to the control plane nodes. See the _Load balancing requirements for user-provisioned infrastructure_ section for more information.
```
Comment (jeana-redhat) [inline modules/install-sno-ibm-z-lpar.adoc:11]:
```suggestion
* If you are deploying a single-node cluster there are zero compute nodes, the Ingress Controller pods run on the control plane nodes. In single-node cluster deployments, you must configure your application ingress load balancer to route HTTP and HTTPS traffic to the control plane nodes. See the _Load balancing requirements for user-provisioned infrastructure_ section for more information.
```

File: modules/install-sno-ibm-z-lpar.adoc [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+
+. Set the host architecture by running the following command:
++
+[source,terminal]
+----
+$ ARCH=<architecture> <1>
+----
+<1> Replace `<architecture>` with the target host architecture `s390x`.
+
+. Download the {product-title} client (`oc`) and make it available for use by entering the following commands:
```
Comment (jeana-redhat) [inline modules/install-sno-ibm-z-lpar.adoc:32]: I see that this is following the model of the existing content. It would be good to update all three to use a single command per step, but that is out of scope here. Suggest adding it as a backlog docs enhancement.

File: modules/install-sno-ibm-z-lpar.adoc [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+----
+$ tar zxvf oc.tar.gz
+----
++
+[source,terminal]
+----
+$ chmod +x oc
+----
+
+. Download the {product-title} installer and make it available for use by entering the following commands:
```
Comment (jeana-redhat) [inline modules/install-sno-ibm-z-lpar.adoc:49]: Guessing this would need to change in all three, can address later if that's better.

```suggestion
. Download the {product-title} installation program and make it available for use by entering the following commands:
```

**Review by jeana-redhat — APPROVED** [review]
This LGTM. I had some suggestions but none need to be addressed before merging. Labeling `ok-to-merge` so you can decide if you want to make any changes or not but it doesn't need any further merge review if you want to go ahead.

/remove-label merge-review-in-progress
/remove-label merge-review-needed
/label ok-to-merge

File: modules/install-sno-ibm-z-lpar.adoc (hunk 13 above)
Comment (SNiemann15) [reply modules/install-sno-ibm-z-lpar.adoc:32]: Will add this and the next comment to the post GA backlog as this applies to x86 procedures too.
